src/sb-util.ts

```diff
@@ -53,6 +54,9 @@ export class ScratchProject {
     }
 
     public sprites(selector?: string): SpriteCollection {
+        this.projectJSON.targets.forEach((p): void => {
```

Comment:
I see where you're going with this. A Sprite will carry around a reference to the project

Reply:
I like this approach and this is exactly what I was thinking would work best for having a project adjacent queryable property. Nice work!

Reply:
Can we make this part of the iterable or otherwise on demand? Adding a non-iterable workload is a slippery slope. If we have too many of these we'll have a slow system. Scratch VM and GUI are in a sense fast but have enough small always-executed workloads that some actions in them are very slow when they don't need to be.

Reply:
Yeah - You'll note I did make it part of the iterable for blocks, but for this one, I kinda hack-jobbed it on the first draft. 

I think once I have a better idea of how much meta-interconnectivity this has, Ill probably re-write this as a `const setMeta = (p) => { getSpriteMeta(p).project = this; return p; }` and some sort of map on the iterable in a second draft